### PR TITLE
Remove model_ids from agent_base and model_base

### LINKED ISSUE #464: [core] `model_ids` should not be necessary in agent_base and algorithm_base
1. move implementation of `set_weights` and `get_weights` from algorithm_base.py to core/fluid/algorithm.py
2. remove `model_ids` from agent_base.py

### PR DESCRIPTION
**Pull request by rical730**
fix https://github.com/PaddlePaddle/PARL/issues/464

### CODE REVIEW

File: parl/core/agent_base.py [hunk 1]
```diff
@@ -27,54 +27,21 @@ def __init__(self, algorithm):
...
-
-         Args:
-            model_ids (List/Set): list/set of model_id, will only return weights of models
-                              whiose model_id in the `model_ids`.
         
         Returns:
             (Dict): Dict of weights ({attribute name: numpy array/List/Dict})
         """
-        return self.alg.get_weights(model_ids=model_ids)
+        return self.alg.get_weights(*args, **kwargs)
```
Comment (TomorrowIsAnOtherDay) [inline parl/core/agent_base.py:36]: Excellent.

File: parl/core/fluid/agent.py [hunk 2]
```diff
@@ -112,6 +112,55 @@ def build_program(self):
         """
         raise NotImplementedError
 
+    # def get_weights(self, model_ids=None):
```
Comment (TomorrowIsAnOtherDay) [inline parl/core/fluid/agent.py]: Please remove the commented lines.

Comment (rical730) [reply parl/core/fluid/agent.py]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
Looks good to me.

**Pull request merged**